pkg/query-service/app/http_handler.go

```diff
@@ -2192,6 +2193,130 @@ func (aH *APIHandler) WriteJSON(w http.ResponseWriter, r *http.Request, response
 	w.Write(resp)
 }
 
+// Preferences
+func (ah *APIHandler) RegisterPreferenceRoutes(router *mux.Router, am *AuthMiddleware) {
+	subRouter := router.PathPrefix("/api/v1").Subrouter()
+
+	// user actions
+	subRouter.HandleFunc("/user/preferences", am.ViewAccess(ah.getAllUserPreferences)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/user/preferences/{preferenceId}", am.ViewAccess(ah.getUserPreference)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/user/preferences/{preferenceId}", am.ViewAccess(ah.updateUserPreference)).Methods(http.MethodPost)
+
+	// org actions
+	subRouter.HandleFunc("/org/preferences", am.AdminAccess(ah.getAllOrgPreferences)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/org/preferences/{preferenceId}", am.AdminAccess(ah.getOrgPreference)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/org/preferences/{preferenceId}", am.AdminAccess(ah.updateOrgPreference)).Methods(http.MethodPost)
+}
+
+func (ah *APIHandler) getUserPreference(
+	w http.ResponseWriter, r *http.Request,
+) {
+	preferenceId := mux.Vars(r)["preferenceId"]
+	orgId := r.URL.Query().Get("orgId")
```

Comment:
The route for this is /user/preferences. It makes more sense to change the route to /org/{orgId}/user/preferences if you always need the orgId to be there.

Otherwise you will have to validate whether the mandatory orgId is present or not here.

Reply:
this code will be changed, we will be relying on the orgID from the JWT token. so orgId will not be the part of the request altogether.